src/project-generator.ts

```diff
@@ -51,266 +39,97 @@ export class ProjectGenerator {
         try {
-            await axios.post(
-                `${PYTHON_TEMPLATE_URL}/generate`,
-                {
-                    owner: this.owner,
-                    name: this.repo,
-                    description: DEFAULT_REPOSITORY_DESCRIPTION,
-                    include_all_branches: false,
-                    private: true,
-                },
-                this.requestConfig
-            );
+            let decodedVspecString = decode(vspecUri);
+            await this.gitRequestHandler.generateRepo();
+            // Delay is introduced to make sure that the git API creates
+            // everything we need before doing other API requests
+            await delay(MS_TO_WAIT_FOR_GITHUB);
+            await this.updateContent(appName, codeSnippet, decodedVspecString);
+            return StatusCodes.OK;
         } catch (error) {
-            if (axios.isAxiosError(error)) {
-                throw new ProjectGeneratorError(error);
-            } else {
-                throw error;
-            }
+            throw error;
         }
-        const responseStatus = await this.checkRepoAvailability();
-        await this.enableWorkflows(false);
-        return responseStatus;
     }
 
-    private async checkRepoAvailability(): Promise<number> {
-        let retries = 0;
-        let success = false;
-        let responseStatus: StatusCodes = StatusCodes.INTERNAL_SERVER_ERROR;
-        const maxRetries = 20;
+    private async updateContent(appName: string, codeSnippet: string, vspec: string): Promise<number> {
+        let vspecJsonBlobSha: string = '';
 
-        while (retries < maxRetries && !success) {
-            try {
-                const response = await axios.get(`${this.repositoryPath}/contents`, this.requestConfig);
-                responseStatus = response.status;
-                success = true;
-                return responseStatus;
-            } catch (error) {
-                if (axios.isAxiosError(error)) {
-                    console.log(`Check #${retries + 1} if Repository is generated failed. Retrying.`);
-                    responseStatus = error.response?.status as number;
-                } else {
-                    throw error;
-                }
-            }
-            retries++;
-        }
-        return responseStatus;
-    }
+        const appManifestBlobSha = await this.getNewAppManifestSha(appName, vspec);
+        const mainPyBlobSha = await this.getNewMainPySha(appName, codeSnippet);
 
-    private async enableWorkflows(isEnabled: boolean): Promise<boolean> {
-        try {
-            await axios.put(
-                `${this.repositoryPath}/actions/permissions`,
-                {
-                    enabled: isEnabled,
-                },
-                this.requestConfig
-            );
-            return true;
-        } catch (error) {
-            console.log(error);
-            return false;
+        if (this.vspecIsPayload(vspec)) {
```

Comment:
The information whether `vspec` is a payload or not is available in all calling methods. I recommend we do not forget this information ant try to retrieve it with an additional method. 

It should be called as `updateContent(appName, codeSnipped, decodedUri)` by  `runWithUri` and `updateContent(appName, codeSnippet, './app/vspec.json')` by `runWithPayload`. The later should also take care of creating the blob rather than doing it in here to let them share unified logic.